src/overlays/actors/ovl_En_Po_Composer/z_en_po_composer.c

```diff
@@ -100,88 +128,612 @@ static DamageTable D_80BC67E8 = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/func_80BC5830.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/func_80BC5898.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/func_80BC58E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/func_80BC59BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/func_80BC59EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/func_80BC5CC8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/func_80BC5D34.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/func_80BC5DEC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/func_80BC5FE8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/EnPoComposer_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/func_80BC60BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/func_80BC617C.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(targetArrowOffset, 3200, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Po_Composer/EnPoComposer_Draw.s")
+static s32 sPlayerIsPlayingOcarina = false;
+
+void EnPoComposer_Init(Actor* thisx, PlayState* play) {
+    EnPoComposer* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 30.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gPoeComposerSkel, &gPoeComposerIdleAnim,
+                       this->jointTable, this->morphTable, PO_COMPOSER_LIMB_MAX);
+    Collider_InitJntSph(play, &this->lanternCollider);
+    Collider_SetJntSph(play, &this->lanternCollider, &this->actor, &sJntSphInit, this->lanternColliderElements);
+    Collider_InitCylinder(play, &this->bodyCollider);
+    Collider_SetCylinder(play, &this->bodyCollider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->lightNode = LightContext_InsertLight(play, &play->lightCtx, &this->lightInfo);
+    Lights_PointGlowSetInfo(&this->lightInfo, this->actor.home.pos.x, this->actor.home.pos.y, this->actor.home.pos.z,
+                            255, 255, 255, 0);
+    this->actor.flags &= ~ACTOR_FLAG_TARGETABLE;
+    this->lightColor = sLightColorInit;
+    this->envColor = sEnvColorInit;
+    this->lastCsAction = COMPOSER_CS_ACTION_NONE;
+
+    if (PO_COMPOSER_IS_FLAT(&this->actor)) {
+        this->sharpCsNum = 0;
+        this->inCutscene = true;
+        EnPoComposer_SetupStartedCutscene(this);
+    } else if (this->actor.params & 0x4000) {
+        if (gSaveContext.save.entrance != 0x20A0) {
+            Actor_Kill(&this->actor);
+        } else {
+            this->sharpCsNum = SHARP_CS_ENCOUNTER_FIRST;
+            this->inCutscene = true;
```

Comment:
```suggestion
            Actor_Kill(&this->actor);
            return;
        }
        this->sharpCsNum = SHARP_CS_ENCOUNTER_FIRST;
        this->inCutscene = true;
```
I think this can be an early return?